src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp

```diff
@@ -534,6 +539,9 @@ class DiscoveryDataBase
     // Whether the database is enabled
     std::atomic<bool> enabled_;
 
+    // Whether it has been a new entity discovered or updated in this subroutine loop
+    std::atomic<bool> entity_updates_;
```

Comment:
I think this could be substituted with a counter that returns the number of created entities since the last time that is was consulted. This way, `PDPServer` is the one deciding whether the DATAs should be sent. Doing this implies changing the getter to reset the counter and return the value that it had, doing a `++`in the `DiscoveryDataBase` in every place that there is currently a `.store(true)`, always return the `pdp_to_send` as dirty topics outputs it, and deciding whether to send or not in `PDPServer`